### PR TITLE
Geolocation hook and wip component

### PR DESCRIPTION
**Pull request by gigxz**
## Description

Proof-of-concept component for requesting geolocation using [Geolocation API](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API).

I wired this up for testing it out, figured we might as well include it in the codebase. Currently it's only used in the Story.

Related issues: https://github.com/open-path/Green-River/issues/5726, https://github.com/open-path/Green-River/issues/5896
## Type of change

[//]: # 'remove options that are not relevant'
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Review by martha — APPROVED** [review]
Cool!
May I ask how you tested this locally? Even with VPN turned off, I'm not finding the location component gets my coordinates accurately, but, neither does Google Maps.

File: src/hooks/useGeolocation.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          break;
+        default:
+          setError('An unknown error occurred.');
+          break;
+      }
+    };
+
+    if (navigator.geolocation) {
+      setLoading(true);
+      navigator.geolocation.getCurrentPosition(handleSuccess, handleError);
```
Comment (ttoomey) [inline src/hooks/useGeolocation.tsx:38]: I'm not familiar with this api but [looks like there's a third options param](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition). For example:
```
const options = {
  timeout: 30 * 1000, // timeout after 30 seconds
  maximumAge: 5 * 60 * 1000, // use a cached result from 5 minutes ago
  // enableHighAccuracy: true
};
navigator.geolocation.getCurrentPosition(handleSuccess, handleError, options);

```

File: src/components/elements/input/GeolocationInput.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          <img
+            alt='Map Holder'
+            src={`https://maps.googleapis.com/maps/api/staticmap?center=${location.latitude + ',' + location.longitude}&zoom=16&size=400x300&output=embed`}
+          />
+        )} */}
+        {coordinates && (
+          <Typography>
+            Latitude: {coordinates.latitude}
+            <br />
+            Longitude: {coordinates.longitude}
```
Comment (ttoomey) [inline src/components/elements/input/GeolocationInput.tsx:36]: could show accuracy too? Maybe a threshold for usable accuracy (like 500 meters)
```
Accuracy: {coordinates.accuracy}
```

**Review by ttoomey — APPROVED** [review]
Looks good! 

I guess the plan is to use this in combo with a map, so if your geolocation isn't working, or you're back filling data from another location you could drop a pin or enter an address?

**Comment by gigxz** [comment]
> May I ask how you tested this locally? Even with VPN turned off, I'm not finding the location component gets my coordinates accurately, but, neither does Google Maps.

I tested by running storybook and clicking "Request location" on the GelocationInput component. It requests location from my browser, and then shows the latitude and longitude on the page. That's interesting, I'm not sure why you might be getting inaccurate coordinates-- but its definitely the case that this api is more accurate on mobile, so maybe that has something to do with it.


> I guess the plan is to use this in combo with a map, so if your geolocation isn't working, or you're back filling data from another location you could drop a pin or enter an address?

Yes on no, may be as a phase 2 for so. [some more context here](https://docs.google.com/document/d/1mlh91wDHOZ81w5RFYRERYP8izXvPRMqWq9cf9XZwkS8/edit#heading=h.83tgztf3bii9)

File: src/components/elements/input/GeolocationInput.tsx (hunk 2 above)
Comment (gigxz) [reply src/components/elements/input/GeolocationInput.tsx:36]: We're probably going to want to show a map here, not the coordinates, this is just a placeholder!

**Pull request merged**